packages/dids/src/methods/did-dht.ts

```diff
@@ -805,10 +805,36 @@ export class DidDhtDocument {
     const publicKeyBytes = DidDhtUtils.identifierToIdentityKeyBytes({ didUri });
 
     // Retrieve the signed BEP44 message from a DID DHT Gateway or Pkarr relay.
-    const bep44Message = await DidDhtDocument.pkarrGet({ gatewayUri, publicKeyBytes });
+    let bep44Message = await DidDhtDocument.pkarrGet({ gatewayUri, publicKeyBytes });
 
     // Verify the signature of the BEP44 message and parse the value to a DNS packet.
-    const dnsPacket = await DidDhtUtils.parseBep44GetMessage({ bep44Message });
+    let dnsPacket = await DidDhtUtils.parseBep44GetMessage({ bep44Message });
+
+    // Look at the NS records in the DNS packet to find the resolution gateway URIs.
+    let resolutionGatewayUris = await DidDhtDocument.getAuthoritativeGatewayUris({ didUri, dnsPacket });
+
+    // Only do a second retrieval if the authoritative resolution gateway URIs are different from the given gateway URI.
+    if(!resolutionGatewayUris.includes(gatewayUri)) {
```

Comment:
This is confusing, the intent seems to only enter the condition check if there is gateway URI specified and the given URI is not in that list. But right now the code inside is still executed when the array is empty.

Reply:
I have a small refactoring PR that address this:

https://github.com/TBD54566975/web5-js/pull/608

Reply:
thanks for catching! I merged it into this branch